### PR TITLE
Create JavaScript loader block in base template

### PR DESCRIPTION
**Pull request by anselmbradford**
Apps (such as owning-a-home) will likely have their own template that overrides the base template's `javascript` block (see [example](https://github.com/cfpb/cfgov-refresh/pull/3773/files#diff-3d9dc87803dad1af3ead8842353b8a87R20)). The problem with this is if we want to use the JS loader that's in the base template it will need to be re-created in each child template. This PR creates a `javascript_loader` block that adds a javascript loader function to the page before the `javascript` block, so that child templates can choose to only override the `javascript` block and utilize the JS loader in the parent template. 

## Additions

- Creates `javascript_loader` block that contains a JS loader function.

## Testing

1. Blog page should still show common.js, email-signup.js, and filterable-list-controls.js in the JS network tab in dev tools.

### CODE REVIEW

**Comment by sebworks** [comment]
The only issue is that you always have to call super because the common.js stuff is in the block.

**Comment by sebworks** [comment]
Also, if you remove the function then the s variable has global scope.

**Comment by anselmbradford** [comment]
@sebworks Okay, I re-added the function. It's in use in https://github.com/cfpb/cfgov-refresh/pull/3818

**Comment by sebworks** [comment]
Ship it

File: cfgov/jinja2/v1/_layouts/base.html [hunk 1]
```diff
@@ -256,7 +264,7 @@
   if ( document.body.parentElement.className.indexOf( 'no-js' ) === -1 ) {
     !function(){
     {# Include site-wide JavaScript. #}
-    var a = [ '{{ static('js/routes/common.js') }}' ];
+    var s = [ '{{ static('js/routes/common.js') }}' ];
```
Comment (Scotchester) [inline cfgov/jinja2/v1/_layouts/base.html]: Minor: should be indented one more level.

File: cfgov/jinja2/v1/_layouts/base.html [hunk 2]
```diff
@@ -295,11 +303,11 @@
...
     {% set js_source = page_js() | trim %}
     {% if js_source | length > 0 %}
-      a.push( '{{ static('js/routes/' + request.path[1:] + 'index.js') }}' );
+      s.push( '{{ static('js/routes/' + request.path[1:] + 'index.js') }}' );
     {% endif %}
 
-    {# Minified dynamic JavaScript loader that injects a script tag in the head of the page #}
-    for(var b,c,d=[],f=document.scripts[0];b=a.shift();)c=document.createElement("script"),"async"in c?(c.async=!1,c.src=b,document.head.appendChild(c)):(d.push(c),c.onreadystatechange=function(){for(var e;d[0]&&"loaded"==d[0].readyState;)e=d.shift(),e.onreadystatechange=null,f.parentNode.insertBefore(e,f)},c.src=b);
+    {# Pass scripts to JavaScript loader. #}
+    jsl(s);
```
Comment (Scotchester) [inline cfgov/jinja2/v1/_layouts/base.html]: Minor: should be indented one more level.

File: cfgov/jinja2/v1/_layouts/base.html (hunk 1 above)
Comment (anselmbradford) [reply cfgov/jinja2/v1/_layouts/base.html]: Fixed

File: cfgov/jinja2/v1/_layouts/base.html (hunk 2 above)
Comment (anselmbradford) [reply cfgov/jinja2/v1/_layouts/base.html]: Fixed

**Comment by anselmbradford** [comment]
Everyone blame @Scotchester 's shoddy review if this doesn't work 😜

**Pull request merged**